en/on-prem-model-service.md

```diff
@@ -0,0 +1,210 @@
+
+```bash
+$ sudo p11-kit server --provider /usr/lib/x86_64-linux-gnu/opensc-pkcs11.so "pkcs11:model=PKCS%2315%20emulated;manufacturer=www.CardContact.de;serial=DENK0300972" -n "/var/snap/snap-store-proxy/common/pkcs11" -f
+
+P11_KIT_SERVER_ADDRESS=unix:path=/var/snap/snap-store-proxy/common/pkcs11; export P11_KIT_SERVER_ADDRESS;
+P11_KIT_SERVER_PID=26963; export P11_KIT_SERVER_PID;
+```
+
+Restart the Model Service (this needs to be done each time the p11-kit server is restarted):
+
+```bash
+snap restart snap-store-proxy.snapmodels
+```
+
+### Snap Store Proxy configuration
+
+Set the HSM label and pin in the Proxy snap:
+
+```bash
+$ p11tool --list-tokens
+
+Token 0:
+    Label: SmartCard-HSM (UserPIN)
+...
+
+$ sudo snap-proxy config proxy.hsm.token-label="SmartCard-HSM (UserPIN)"
+
+$ sudo snap-proxy config proxy.hsm.token-pin=74656
+```
+
+## Model Service CLI Usage
+
+The Model Service management CLI is provided by the `store-admin` snap.
+
+### Create a signing key on the HSM
+
+Create a signing key using `store-admin` for signing serial requests:
+
+```bash
+$ BRAND_ACCOUNT_ID=<brand-account-id> store-admin create key test-key
+Generating a signing keypair on the proxy's HSM. This may take some time.
+Signing key 'test-key' created.
+
+$ store-admin list keys
+Name      SHA3-384
+--------  ----------------------------------------------------------------
+test-key  PPkB6XcYjkxzA9c6dXsaM0sg9r_d5DZ2kDYvWPTeuSXofXGzMDBt7DoD_Xiw3see
+```
+
+The `BRAND_ACCOUNT_ID` environment variable only needs to be set once; it will be stored and automatically used subsequently.
+
+The key needs to be registered with the online Snap Store before it can sign serials:
+
+```bash
+$ store-admin register-key PPkB6XcYjkxzA9c6dXsaM0sg9r_d5DZ2kDYvWPTeuSXofXGzMDBt7DoD_Xiw3see
+Registering signing key with the global Snap Store...
+...
+
+Key PPkB6XcYjkxzA9c6dXsaM0sg9r_d5DZ2kDYvWPTeuSXofXGzMDBt7DoD_Xiw3see registered.
+```
+
+The account-key assertion needs to be pushed to the air-gapped Proxy. First, export the assertion:
+
+```bash
+snap known --remote account-key public-key-sha3-384=PPkB6XcYjkxzA9c6dXsaM0sg9r_d5DZ2kDYvWPTeuSXofXGzMDBt7DoD_Xiw3see > test-key.assert
+```
+
+Copy the assertion to the Snap Store Proxy's `$SNAP_COMMON` directory on the Proxy host, then push the assertion to the Proxy:
+
+```bash
+sudo snap-proxy push-account-keys /var/snap/snap-store-proxy/common/test-key.assert
+```
+
+### Add a model in the Model Service
+
+To sign the serial requests for devices of a model, the model name as configure in the Model Service must match that in the model assertion:
+
+```bash
+$ store-admin create model model-a
+API key (alphanumeric, `pwgen 40` for options): model-a-apikey
+Model 'model-a' created.
+```
+
+!!! Neutral "Note":
+    The account-key assertion for the key used to sign the model assertion must also be pushed to the Proxy.
```

Comment:
Thoughts on moving this up to where push-account-keys command above is? And maybe give an example command for it too? Or mention that it could have been done as part of store import.

Reply:
Thanks, I've expanded the store import section to mention that keys existing at the time of store export should be included in the bundle (using the `--key` flag) if they were used to sign model assertions. 

I've also added a note to the `push-account-keys` section above, mentioning that new keys used for model assertions need to be pushed separately, if any are created after the store bundle import.